gitbook/audio-streams.md

```diff
@@ -2,143 +2,129 @@
+        data.sample_count = ... # put the number of audio samples available in the new chunk */;
+        true
+    end
+end
 ```
 
-You must return `true` when everything is all right, or `false` if playback must be stopped, either because an error has occurred or because there's simply no more audio data to play. 
+You must return `true` when everything is all right, or `false` if playback must be stopped, either because an error has occurred or because there's simply no more audio data to play.
 
-SFML makes an internal copy of the audio samples as soon as `onGetData` returns, so you don't have to keep the original data alive if you don't want to. 
+CrSFML makes an internal copy of the audio samples as soon as `on_get_data` returns, so you don't have to keep the original data alive if you don't want to.
 
-The `onSeek` function is called when the `setPlayingOffset` public function is called. Its purpose is to change the current playing position in the source data. The parameter is a time value representing the new position, from the beginning of the sound (*not* from the current position). This function is sometimes impossible to implement. In those cases leave it empty, and tell the users of your class that changing the playing position is not supported. 
+The `on_seek` function is called when the `playing_offset=` public function is called. Its purpose is to change the current playing position in the source data. The parameter is a time value representing the new position, from the beginning of the sound (*not* from the current position). This function is sometimes impossible to implement. In those cases leave it empty, and tell the users of your class that changing the playing position is not supported.
 
-Now your class is almost ready to work. The only thing that [SoundStream]({{book.api}}/SoundStream.html) needs to know now is the channel count and sample rate of your stream, so that it can be played as expected. To let the base class know about these parameters, you must call the `initialize` protected function as soon as they are known in your stream class (which is most likely when the stream is loaded/initialized). 
+Now your class is almost ready to work. The only thing that [SoundStream]({{book.api}}/SoundStream.html) needs to know now is the channel count and sample rate of your stream, so that it can be played as expected. To let the base class know about these parameters, you must call the `initialize` protected function as soon as they are known in your stream class (which is most likely when the stream is loaded/initialized).
 
-```
-// where this is done totally depends on how your stream class is designed
-unsigned int channelCount = ...;
-unsigned int sampleRate = ...;
-initialize(channelCount, sampleRate);
+```ruby
+# where this is done totally depends on how your stream class is designed
+channel_count = ...
+sample_rate = ...
+initialize(channel_count, sample_rate)
```

Comment:
`SoundStream#initialize` does not exist, and if it did, it would conflict with Crystal constructor method name